xtask/src/cargo.rs

```diff
@@ -48,6 +48,25 @@ impl Package {
     /// Update the version of this crate in dependant crates' manifests, with the given version
     /// prefix.
     pub(crate) fn update_dependants(&self, metadata: &Metadata, dry_run: bool) -> Result<()> {
+        let workspace_manifest_path = metadata.workspace_root.join("Cargo.toml");
+        let mut document = read_file(&workspace_manifest_path)?.parse::<Document>()?;
+        let workspace_deps = &mut document["workspace"]["dependencies"];
+
+        if workspace_deps.get(&self.name).is_some() {
+            println!("Updating workspace dependency…");
+            if !dry_run {
+                let version = if self.name == "ruma-macros" || !self.version.pre.is_empty() {
+                    format!("={}", self.version)
+                } else {
+                    self.version.to_string()
+                };
+
+                workspace_deps[&self.name]["version"] = value(version.as_str());
+
+                write_file(&workspace_manifest_path, document.to_string())?;
+            }
+        }
+
         for package in metadata.packages.iter().filter(|p| {
```

Comment:
Isn't this unnecessary if all crates are workspace dependencies?

In any case this should check if this is a workspace dependency before trying to change the version.